tests/framework/test_time.py

```diff
@@ -114,16 +104,66 @@ def test_unequal_steps(SimulationContext, base_config, components):
     # Do a step just to update the next_event_time
     sim.step()
     # Check that next step, we will still update all
-    assert (
-        len(
-            sim._clock.get_active_population(
-                sim.get_population().index,
-                sim._clock.event_time,
-            )
-        )
-        == pop_size
-    )
-    # Check that we actually include 7 in events
+    assert len(active_simulants()) == pop_size
+    # Check that we actually include simulant of index 7 in events
     sim.step()
     for index in listener.event_indexes.values():
         assert 7 in index
+
+
+def test_step_pipeline(SimulationContext, base_config, components):
+    sim = SimulationContext(base_config, components)
+    sim.setup()
+    sim.initialize_simulants()
+    pop_size = len(sim.get_population())
+    active_simulants = lambda: sim._clock.get_active_population(
+        sim.get_population().index,
+        sim._clock.event_time,
+    )
+
+    pipeline = lambda: sim._values.get_value("simulant_step_size")(
+        sim._population.get_population(True).index
+    )
+
+    column = lambda: sim._population._population.step_size
+
+    assert np.all(pipeline() == column())
+    assert len(active_simulants()) == pop_size
+
+    sim.step()
+    assert np.all(pipeline() == column())
+    assert len(active_simulants()) == pop_size
+
+
+@pytest.mark.parametrize("step_modifier", [0.5, 1, 2, 3.5, 5])
+def test_step_pipeline_with_modifier(SimulationContext, base_config, step_modifier):
+    sim = SimulationContext(base_config, [StepModifier("step_modifier", step_modifier)])
+    sim.setup()
+    sim.initialize_simulants()
+    pop_size = len(sim.get_population())
+    active_simulants = lambda: sim._clock.get_active_population(
+        sim.get_population().index,
+        sim._clock.event_time,
+    )
+    pipeline = lambda: sim._values.get_value("simulant_step_size")(
+        sim._population.get_population(True).index
+    )
+    column = lambda: sim._population._population.step_size
+
+    ## Everyone starts active
+    assert np.all(pipeline() == column())
+    assert len(active_simulants()) == pop_size
+
+    ## Go through a couple simulant step cycles
+    for _ in range(2):
+        for _ in range(math.ceil(step_modifier) - 1):
```

Comment:
Is 3.5 effectively the same thing as 4?

Reply:
Yeah, this is something that needs to get dealt with in the next ticket (MIC-4638), since we update steps based on the global clock, but update pipeline values based on the simulant step size, even if that doesn't conform to to the _actual_ number of steps that we took. I didn't totally appreciate that until this comment, so thanks!